TypeCobol.LanguageServer/TypeCobolServer.cs

```diff
@@ -239,6 +239,18 @@ protected DocumentContext GetDocumentContextFromStringUri(string uri, Workspace.
             return null;
         }
 
+        /// <summary>
+        /// Gets the original URI (which was set by the client)
+        /// DON'T use ToString() as it returns the canonically unescaped form of the URI
+        /// (it may cause issue if the path contains some blanks which need to be escaped)
+        /// </summary>
+        /// <param name="uri"></param>
+        /// <returns></returns>
+        private static string GetOriginalUri(Uri uri)
+        {
+            return uri.OriginalString;
+        }
```

Comment:
Please move that to https://github.com/TypeCobolTeam/TypeCobol/blob/develop/TypeCobol.LanguageServer/Utilities/Extensions.cs and make it extension an extension method.

Reply:
I think it's better to create a new constructor for `TextDocumentIdentifier` accepting an `Uri` parameter and call `OriginalString` in it. Duplicate the call to `OriginalString` for the `diagParameter.uri` assignment and remove this method.
You can keep the comment about using the original value sent by the client in the new constructor.